src/api/app/controllers/webui/packages/bs_requests_controller.rb

```diff
@@ -27,34 +28,23 @@ def index
 
       private
 
-      def filter_by_involvement(involvement)
-        return filter_by_involvement_staging_project(involvement) if staging_projects.present?
+      def set_bs_requests
+        return unless Flipper.enabled?(:request_index, User.session)
 
-        case involvement
-        when 'all'
-          target = BsRequest.with_actions_and_reviews.where(bs_request_actions: { target_project: @project.name, target_package: @package.name })
-          source = BsRequest.with_actions_and_reviews.where(bs_request_actions: { source_project: @project.name, source_package: @package.name })
-          reviews = BsRequest.with_actions_and_reviews.where(reviews: { package: @package })
-          target.or(source).or(reviews)
-        when 'incoming'
-          BsRequest.with_actions.where(bs_request_actions: { target_project: @project.name, target_package: @package.name })
-        when 'outgoing'
-          BsRequest.with_actions.where(bs_request_actions: { source_project: @project.name, source_package: @package.name })
-        end
+        @bs_requests = @package.bs_requests
       end
 
-      def filter_by_involvement_staging_project(involvement)
-        case involvement
-        when 'all'
-          target = BsRequest.with_actions_and_reviews.where(staging_project: staging_projects, bs_request_actions: { target_project: @project.name, target_package: @package.name })
-          source = BsRequest.with_actions_and_reviews.where(staging_project: staging_projects, bs_request_actions: { source_project: @project.name, source_package: @package.name })
-          reviews = BsRequest.with_actions_and_reviews.where(staging_project: staging_projects, reviews: { package: @package })
-          target.or(source).or(reviews)
-        when 'incoming'
-          BsRequest.with_actions.where(staging_project: staging_projects, bs_request_actions: { target_project: @project.name, target_package: @package.name })
-        when 'outgoing'
-          BsRequest.with_actions.where(staging_project: staging_projects, bs_request_actions: { source_project: @project.name, source_package: @package.name })
-        end
+      def filter_involvement
+        return if params[:involvement]&.compact_blank.blank?
+
+        @selected_filter['involvement'] = params[:involvement]
+        @bs_requests = case
+                       when @selected_filter['involvement'].include?('incoming')
+                         @bs_requests.where(bs_request_actions: { target_package_id: @package.id })
+                                     .or(@bs_requests.where(reviews: { by_project: @project.name, by_package: @package.name }))
```

Comment:
You could apply your fix of "switching away from strings to user_id/project_id/package_id" here, too, if I'm not wrong...

Reply:
done